packages/core/src/middleware/state.ts

```diff
@@ -87,6 +87,13 @@ export interface SnapshotableContext {
   ) => () => void;
 }
 
+export type StatefulContext<
```

Comment:
state.ts must extend the type of the context passed to a handler, so we don't need `context as StatefulContext<AgentWorkflowState>`